compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -857,6 +884,12 @@ private BType getFiniteTypeWithValuesOfSingleType(BUnionType unionType, BType ma
     }
 
     private BType getIntLiteralType(BType expType, Object literalValue) {
+        // The literalValue will be a string if it is not within the bounds of what is supported by Java Long,
+        // indicating that it is an overflown Ballerina int
+        if (!(literalValue instanceof Long)) {
+            resultType = symTable.semanticError;
+            return resultType;
```

Comment:
Why can't we do `return symTable.semanticError;`

Reply:
we need to set the result type here first else it is not getting set.